src/controller/CHIPPersistentStorageDelegate.h

```diff
@@ -79,6 +79,19 @@ class DLL_EXPORT PersistentStorageDelegate
      */
     virtual void GetKeyValue(const char * key) = 0;
 
+    /**
+     * @brief
+     *   Lookup the key and call delegate object with it's stringified value.
+     *   This is a synchronous API, where the value is returned via the output
```

Comment:
I wonder how feasible it would be to make the things using this async, so we don't need this sync API.  It's a bit of a pain for consumers, but having a blocking API hidden deep within an API boundary is not great either...  Maybe a followup to switch to an async setup here?

Reply:
Yes, I agree about async. This PR was snowballing with all the changes. I'll try it in a follow on PR.

Reply:
Filed this issue: https://github.com/project-chip/connectedhomeip/issues/3738